lib/bolt/project_migrator/inventory.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require 'bolt/project_migrator/base'
+
+module Bolt
+  class ProjectMigrator
+    class Inventory < Base
+      def migrate(inventory_file, backup_dir)
+        inventory_1_to_2(inventory_file, backup_dir)
+      end
+
+      # Migrates an inventory v1 file to inventory v2.
+      #
+      private def inventory_1_to_2(inventory_file, backup_dir)
+        unless File.exist?(inventory_file)
+          return true
+        end
+
+        data = Bolt::Util.read_yaml_hash(inventory_file, 'inventory')
+        data.delete('version') if data['version'] != 2
+        migrated = migrate_group(data)
+
+        return true unless migrated
+
+        @outputter.print_message "Migrating inventory\n\n"
+
+        backup_file(inventory_file, backup_dir)
+
+        if File.write(inventory_file, data.to_yaml)
```

Comment:
`File.write` returns the number of bytes written, which will always be an integer unless an error is raised writing the file. We should try writing and rescue an error instead of this `if...else` statement:
```
lucy ~/githubs/bolt/project ((570719b...)) $ be bolt project migrate
Migrating inventory

  → Backing up inventory.yaml from
    /home/lucy/githubs/bolt/project/inventory.yaml to
    /home/lucy/githubs/bolt/project/.bolt-bak/inventory.yaml.20200924_131921810.bak

bundler: failed to load command: bolt (/home/lucy/githubs/bolt/vendor/bundle/ruby/2.6.0/bin/bolt)
Errno::EACCES: Permission denied @ rb_sysopen - /home/lucy/githubs/bolt/project/inventory.yaml
  /home/lucy/githubs/bolt/lib/bolt/project_migrator/inventory.rb:29:in `write'
  /home/lucy/githubs/bolt/lib/bolt/project_migrator/inventory.rb:29:in `inventory_1_to_2'
  /home/lucy/githubs/bolt/lib/bolt/project_migrator/inventory.rb:9:in `migrate'
  /home/lucy/githubs/bolt/lib/bolt/project_migrator.rb:48:in `migrate_inventory'
  /home/lucy/githubs/bolt/lib/bolt/project_migrator.rb:15:in `migrate'
  /home/lucy/githubs/bolt/lib/bolt/cli.rb:465:in `execute'
  /home/lucy/githubs/bolt/exe/bolt:11:in `<top (required)>'
  /home/lucy/githubs/bolt/vendor/bundle/ruby/2.6.0/bin/bolt:23:in `load'
  /home/lucy/githubs/bolt/vendor/bundle/ruby/2.6.0/bin/bolt:23:in `<top (required)>'
```
```